src/setupStore.js

```diff
@@ -2,6 +2,8 @@ import { createStore, applyMiddleware, compose, combineReducers } from "redux";
 import authReducer from "./store/reducers/authReducer";
 import userReducer from "./store/reducers/userReducer";
 import hackReducer from "./store/reducers/hackathonReducer";
+import { persistStore, persistReducer } from 'redux-persist'
```

Comment:
You should also be adding a `<PersistGate>` (https://github.com/rt2zz/redux-persist/blob/master/docs/PersistGate.md) around the entire application (or at lead the authenticated parts).

This will ensure the store has been rehydrated before anything tries to use the token from state to load any data.

Reply:
It also looks like it had a loading function, it will be good to explore that in the future